core/src/main/java/org/infinispan/cache/impl/SimpleCacheImpl.java

```diff
@@ -1090,7 +1092,7 @@ public CompletableFuture<Boolean> replaceAsync(K key, V oldValue, V newValue, Me
 
    @Override
    public CompletableFuture<V> getAsync(K key) {
-      return CompletableFuture.completedFuture(get(key));
+      return blockingManager.supplyBlocking(() -> get(key), "get-async").toCompletableFuture();
```

Comment:
I actually thought of a much cleaner way of doing this. Instead, we should use peek on the data container and then do the expiration check ourselves for the SimpleCacheImpl. This way we can return synchronously when not expired and if it is then we can submit the expiration call to the blocking manager.